### PR TITLE
chore: upgrade flake8

### PR DESCRIPTION
**Pull request by c0state**
We need to update this now to support renovate upgrades of flake8 downstream I think?

### CODE REVIEW

**Review by itajaja — COMMENTED** [review]
yeah, I guess. maybe we could lift the pin from these. ideally the pinning happens in poetry.lock or whatever the lock file is

**Comment by c0state** [comment]
> yeah, I guess. maybe we could lift the pin from these. ideally the pinning happens in poetry.lock or whatever the lock file is

Hmm yeah, I thought we pinned since if we didn't, an install of a given `fourmat` version going forward could result in different flake8 (or other pinned dep) version being used but I guess it's theoretically ok if we just locked the major, assuming the deps in question adhere to version conventions reliably (which these should I think)

File: setup.py [hunk 1]
```diff
@@ -28,7 +28,7 @@
         # Pin these to avoid unplanned messy diffs.
         "black==20.8b1",
         "flake8-bugbear==20.1.4",
```
Comment (c0state) [inline setup.py:30]:
```suggestion
        "black~=20",
        "flake8-bugbear~=21",
```

File: setup.py [hunk 2]
```diff
@@ -28,7 +28,7 @@
...
-        "flake8==3.8.4",
+        "flake8==3.9",
```
Comment (c0state) [inline setup.py:31]:
```suggestion
        "flake8~=3",
```

File: setup.py [hunk 3]
```diff
@@ -28,7 +28,7 @@
...
         "isort~=5.5",
```
Comment (c0state) [inline setup.py:32]:
```suggestion
        "isort~=5",
```

File: setup.py [hunk 4]
```diff
@@ -26,10 +26,10 @@
     install_requires=(
         "click>=7",
         # Pin these to avoid unplanned messy diffs.
```
Comment (c0state) [inline setup.py:28]:
```suggestion
        # Pin these to avoid unplanned messy diffs.
```

File: setup.py [hunk 5]
```diff
@@ -25,11 +25,10 @@
...
         "click>=7",
-        # Pin these to avoid unplanned messy diffs.
-        "black==20.8b1",
-        "flake8-bugbear==20.1.4",
-        "flake8==3.8.4",
-        "isort~=5.5",
+        "black~=20",
+        "flake8-bugbear~=21",
+        "flake8~=3",
+        "isort~=5",
```
Comment (itajaja) [inline setup.py:31]: can you make them > < ranges?

Comment (itajaja) [reply setup.py:31]: not sure how to interpret the tilde here, it seems that it has meaning only if you specify at least two numbers... https://www.python.org/dev/peps/pep-0440/#compatible-release

**Review by sloria — APPROVED** [review]
Seems fine to me. The only one that might still be worth pinning still is black, since it's using calver and still pre-stable.

File: setup.py (hunk 5 above)
Comment (c0state) [reply setup.py:31]: Hmm, you're right it specifically states to not use it with a "single segment version number". Let me use `<, >`, etc.

Comment (c0state) [reply setup.py:31]:
```suggestion
        "black>=20,<21",
        "flake8-bugbear>=21,<22",
        "flake8>=3,<4",
        "isort>=5,<6",
```

File: setup.py [hunk 6]
```diff
@@ -25,11 +25,10 @@
     package_data={"fourmat": ("assets/*.*", "assets/.*")},
     install_requires=(
         "click>=7",
-        # Pin these to avoid unplanned messy diffs.
-        "black==20.8b1",
-        "flake8-bugbear==20.1.4",
-        "flake8==3.8.4",
-        "isort~=5.5",
+        "black>=20,<21",
```
Comment (c0state) [inline setup.py:28]: Revert to pinning black for now as suggested

```suggestion
        "black==20.8b1",
```

**Review by sloria — APPROVED** [review]

**Review by itajaja — APPROVED** [review]

**Pull request merged**